### PR TITLE
  fix missing-prototypes warnings in torch_cpu (Part 4) 

### PR DESCRIPTION
**Pull request by cyyever**
This PR fixes more missing-prototypes violations in the torch_cpu source following PRs #100053, #100147 and #100245

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @EikanWang @albanD @zou3519 @Skylion007

### CODE REVIEW

File: aten/src/ATen/core/ivalue.cpp [hunk 1]
```diff
@@ -763,11 +763,6 @@ IValueComparator getGreaterThanComparator(const IValue& v) {
   };
 }
 
-static std::ostream& operator<<(std::ostream& out, const ivalue::EnumHolder& v) {
```
Comment (albanD) [inline aten/src/ATen/core/ivalue.cpp:766]: This is very nice to have around when doing debugging. Can we keep this please?

**Review by albanD — COMMENTED** [review]
Thanks for the change! Looks really good. I think we do want to keep printing functions for when we're doing debugging. Not sure what is the cleanest way to do that.

**Comment by cyyever** [comment]
I had to remove some functions because the Bazel builds kept complaining about unused function. The current CMake setting of noerror=unused-functions isn't copied in the Bazel version

File: aten/src/ATen/core/ivalue.cpp (hunk 1 above)
Comment (cyyever) [reply aten/src/ATen/core/ivalue.cpp:766]: Recover it

**Review by albanD — APPROVED** [review]
SGTM thanks!

File: aten/src/ATen/native/nested/NestedTensorMatmul.cpp [hunk 2]
```diff
@@ -78,64 +78,6 @@ Tensor bmm_nested(const Tensor& self, const Tensor& mat2) {
...
-// Args:
-//     self_sizes: the sizes of `self` in `matmul_nested`
-//     mat2_sizes: the sizes of `mat2` in `matmul_nested`
-//     buffer_op: the options for new buffer
-//     sizemat_op: the options for new size matrix
-// Returns:
-//     the batch size of each input underlying tensor, i.e. the product of batch-dimension sizes
-//     the empty output nested tensor
-inline std::tuple<std::vector<int64_t>, Tensor>
-matmul_nested_helper(
```
[the remark this answers is not in the document]
Comment (drisspg) [reply aten/src/ATen/native/nested/NestedTensorMatmul.cpp:92]: @cpuhrsch  I think you were looking at this last with using group gemm for bmm?

[the remark this answers is not in the document]
Comment (cpuhrsch) [reply aten/src/ATen/native/nested/NestedTensorMatmul.cpp:92]: This is unused code that we can remove.

**Comment by izaitsevfb** [comment]
Hi @cyyever, I'm terribly sorry, but I had to temporarily revert this and #101788, because they break internal Meta builds with the linker error:
```
 ld.lld: error: undefined symbol: nnc_aten_embedding
 ```
We probably need @albanD's help to shepherd this in, and he's not available today. Let's try again after the weekend.

**Pull request closed without merging**